pkg/apis/pipeline/v1beta1/merge.go

```diff
@@ -174,3 +176,24 @@ func mergeObjWithTemplateBytes(md *mergeData, obj, out interface{}) error {
 	// Unmarshal the merged JSON to a pointer, and return it.
 	return json.Unmarshal(mergedAsJSON, out)
 }
+
+// amendConflictingContainerFields amends conflicting container fields after merge, and overrides conflicting fields
+// by fields in step.
+func amendConflictingContainerFields(container *corev1.Container, step Step) {
```

Comment:
Could you add this change to v1 as well?
https://github.com/tektoncd/pipeline/blob/main/pkg/apis/pipeline/v1/merge.go
Thanks!

Reply:
ok, I have added modifications to v1.